spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -788,68 +788,82 @@ class CometSparkSessionExtensions
           }
 
         // Native shuffle for Comet operators
-        case s: ShuffleExchangeExec
-            if isCometShuffleEnabled(conf) &&
+        case s: ShuffleExchangeExec =>
+          val nativeShuffle: Option[SparkPlan] =
+            if (isCometShuffleEnabled(conf) &&
               isCometNativeShuffleMode(conf) &&
-              QueryPlanSerde.supportPartitioning(s.child.output, s.outputPartitioning)._1 =>
-          logInfo("Comet extension enabled for Native Shuffle")
-
-          val newOp = transform1(s)
-          newOp match {
-            case Some(nativeOp) =>
-              // Switch to use Decimal128 regardless of precision, since Arrow native execution
-              // doesn't support Decimal32 and Decimal64 yet.
-              conf.setConfString(CometConf.COMET_USE_DECIMAL_128.key, "true")
-              val cometOp = CometShuffleExchangeExec(s, shuffleType = CometNativeShuffle)
-              CometSinkPlaceHolder(nativeOp, s, cometOp)
-            case None =>
-              s
-          }
+              QueryPlanSerde.supportPartitioning(s.child.output, s.outputPartitioning)._1) {
+              val newOp = transform1(s)
+              newOp match {
+                case Some(nativeOp) =>
+                  // Switch to use Decimal128 regardless of precision, since Arrow native execution
+                  // doesn't support Decimal32 and Decimal64 yet.
+                  conf.setConfString(CometConf.COMET_USE_DECIMAL_128.key, "true")
+                  val cometOp = CometShuffleExchangeExec(s, shuffleType = CometNativeShuffle)
+                  Some(CometSinkPlaceHolder(nativeOp, s, cometOp))
+                case None =>
+                  None
+              }
+            } else {
+              None
+            }
 
-        // Columnar shuffle for regular Spark operators (not Comet) and Comet operators
-        // (if configured).
-        // If the child of ShuffleExchangeExec is also a ShuffleExchangeExec, we should not
-        // convert it to CometColumnarShuffle,
-        case s: ShuffleExchangeExec
-            if isCometShuffleEnabled(conf) && isCometJVMShuffleMode(conf) &&
```

Comment:
Hmm, if it is auto mode, why this is not triggered?

Reply:
This part is triggered, but in the following code, `transform1(s)` returns `None` so we return the original shuffle.

```scala
          val newOp = transform1(s)
          newOp match {
            case Some(nativeOp) =>
              // Switch to use Decimal128 regardless of precision, since Arrow native execution
              // doesn't support Decimal32 and Decimal64 yet.
              conf.setConfString(CometConf.COMET_USE_DECIMAL_128.key, "true")
              val cometOp = CometShuffleExchangeExec(s, shuffleType = CometNativeShuffle)
              CometSinkPlaceHolder(nativeOp, s, cometOp)
            case None =>
              s
          }
```

Reply:
This is now explained in more detail in new issue https://github.com/apache/datafusion-comet/issues/1248